pkg/virtctl/convertmachinetype/convert_machine_type.go

```diff
@@ -0,0 +1,136 @@
+package convertmachinetype
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	COMMAND_CONVERT_MACHINE_TYPE = "convert-machine-type"
+)
+
+type Command struct {
+	clientConfig clientcmd.ClientConfig
+	command      string
+}
+
+// holding flag information
+var (
+	namespaceFlag     string
+	forceRestartFlag  bool
+	labelSelectorFlag string
+)
+
+// NewConvertMachineTypeCommand generates a new "convert-machine-types" command
+func NewConvertMachineTypeCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "convert-machine-type",
+		Short: "Perform a mass machine type transition on any VMs that have an outdated machine type.",
+		Long: `Create a Job that iterates through VMs, updating the machine type of any VMs that have an outdated machine type. If a VM is running, it will also label the VM with 'restart-vm-required=true', indicating the user will need to perform manually by default. If --force-restart is set to true, the VM will be automatically restarted and the label will be removed. The Job will terminate once all VMs have their machine types updated, and all 'restart-vm-required' labels have been cleared.
+If no namespace is specified via --namespace, the mass machine type transition will be applied across all namespaces.
+Note that should the Job fail, it will be restarted. Additonally, once the Job is terminated, it will not be automatically deleted. The Job can be monitored and then deleted manually after it has been terminated using 'kubectl'.`,
+		Example: usage(),
+		Args:    templates.ExactArgs("convert-machine-type", 0),
```

Comment:
Why is this needed?

Reply:
This was to be consistent with the implementation of other commands in virtctl. The `version` [command](https://github.com/kubevirt/kubevirt/blob/a26c6c998c5f808bf713ad49d9ebc5e0a9a9bdda/pkg/virtctl/version/version.go#L29) also takes no arguments and is implemented the same way. 

Unless I am misunderstanding this comment and you are referring to the long description, in which case do you feel it is not necessary to explain these things in this description? I was using the `expose` [command](https://github.com/kubevirt/kubevirt/blob/a26c6c998c5f808bf713ad49d9ebc5e0a9a9bdda/pkg/virtctl/expose/expose.go#L54)'s long description as an example to follow.

Reply:
Oh sorry, I was actually referring to `Args:    templates.ExactArgs("convert-machine-type", 0),`